client/src/state/reducer.ts

```diff
@@ -32,18 +33,29 @@ export const reducer = (state: State, action: Action): State => {
 			if (action.payload)
 				return {
 					...state,
-					loggedUser: action.payload
+					loggedUser: action.payload,
+					dictionary: dictionaryList[action.payload.language],
+					themeWithLocale: themeWithLocale(action.payload.language)
 				};
 			else {
 				return { ...state, loggedUser: undefined };
 			}
 		case 'SET_USER_LANGUAGE':
-			return {
-				...state,
-				userLanguage: action.payload,
-				dictionary: dictionaryList[action.payload as keyof typeof dictionaryList],
-				themeWithLocale: themeWithLocale(action.payload)
-			};
+			if (action.payload) {
```

Comment:
Is this check actually needed? It seems that payload in setUserLanguage takes 'LanguageOption' only and can not be undefined per se :)

Reply:
I guess it staid from the previous version :)

Reply:
actually yeah you're right